src/v/storage/types.h

```diff
@@ -302,4 +302,21 @@ struct compaction_config {
 
     friend std::ostream& operator<<(std::ostream&, const compaction_config&);
 };
+
+struct compaction_result {
+    explicit compaction_result(size_t sz)
+      : already_compacted(true)
+      , size_before(sz)
+      , size_after(sz) {}
+
+    compaction_result(size_t before, size_t after)
+      : already_compacted(false)
+      , size_before(before)
+      , size_after(after) {}
+
+    bool already_compacted;
```

Comment:
optionally, could have a bool method that says `bool did_compact() const { return before != after; }`

Reply:
I like this idea, will change